### PR TITLE
Draft foundational skills

### PR DESCRIPTION
**Pull request by jrosen48**
I am hesitant to post and share for a couple of reasons (not ready, not good), but wanted to put this out there to get feedback/request ideas and your input. This is a draft of a section on foundational skills. It's already probably too long, despite also not being ready. Major and minor contributions and feedback are WELCOME. 

A few specific questions I have:

- Do we need a chapter even as long as it is now, given the existence of other excellent resources (i.e., R4DS!)
- Is the idea of downloading a CSV and then reading it useful, useful but a bit tedious, or not useful at this stage/it’s better to get started in some other ways
- Should this chapter include visualizing and modeling data? communicating (w/ R Markdown) too?
- Does it explain basic ideas but also respect readers’ intelligence/expertise in related topics?

### CODE REVIEW

**Comment by jrosen48** [comment]
Wonder if this section should maybe stop at loading data-

**Comment by kierisi** [comment]
This is a great start to foundational skills! To answer your specific questions:

**Do we need a chapter even as long as it is now, given the existence of other excellent resources (i.e., R4DS!)**

- One way to approach this is to provide readers with the minimum amount of information they need to work through the chapter, while pointing them to resources such as R4DS, Modern Dive, STAT545, etc. as needed (for example, as an "if you'd like to know _more_" add-on. 
- Thinking back on the last few years of learning R, it took me awhile to develop a good sense of what's out there in terms of resources. While we all know of R4DS, it could be entirely new to a reader (let alone the idea of there being an entire `tidyverse`.)

**Is the idea of downloading a CSV and then reading it useful, useful but a bit tedious, or not useful at this stage/it’s better to get started in some other ways**

- It likely depends on how users _generally_ encounter data. For me, personally, 99% of the time I **have** to download a .csv file in order to work with the data, as there isn't necessarily an easy way to access the data otherwise. So my personal experience is that downloading a .csv and reading it into `R` is an essential component of the workflow.

**Should this chapter include visualizing and modeling data? communicating (w/ R Markdown) too?**
- One of the ideas I've been toying with (in terms of teaching `R`) is that not very many materials utilize project-based learning, and instead walk the learner through some kind of amorphous set of "things they may encounter." That being said, what if we used the walkthroughs as PBL templates, and collaborated on making the **Foundational Skills** section being one that 1) prepares learners for what they'll encounter in the walkthroughs but also 2) serves as an ongoing reference as the learner works through each of the walkthroughs?

**Does it explain basic ideas but also respect readers’ intelligence/expertise in related topics?**
- Yes! There are some spots of language that can be cleaned up a bit (use of words such as "just" and "simple/simply", for example) but as a whole it does a good job respecting intelligence/expertise.
- One thing to consider is how it's organized - would it work to have two "tracks"? For example, those who have some familiarity with R vs. those who are brand-new?

**Other notes**
- In the beginning sections it should be mentioned that R needs to be installed before RStudio (unless that's changed?) 
- The text refers the reader to CRAN to download RStudio instead of CRAN for R and RStudio for... RStudio
- It's unclear _why_ the reader is being sent to `swirl` or DataCamp, and what they're supposed to do while there. Are they supposed to do an entire track? A video or two? Also, because none of us have enough projects, we _could_ build our own set of lessons in `swirl` 
- There are a lot of options for loading files, but not necessarily explanations as to what to use, when. One of the decisions to be made is perhaps: are we going to draw a line and say "this is our workflow" and provide links/resources when something falls outside of that? Or are we going to try and cover a variety of use cases?

**Comment by restrellado** [comment]
Great work Josh and Isabella! To Jesse's great points, I think there are a lot of interesting possibilities we can choose from here. I think this is a super useful chapter, even if other great resources like R4DS cover similar things. Part 2 of the book is supposed to teach with examples that feel familiar to educators. I think a chapter like this can be welcoming to an educator who's never done this before. More veteran uses will know to jump to the sections that are right for them, but they might even enjoy reading through examples that look more like what they see everyday at work. 

I like that this chapter includes reading in data. I think a lot about the educators who are brand new to R and data science. I think reading in data is a new concept for those coming from a spreadsheet-only workflow, but it's one that is relatively simple and can get the reader a quick win and a nice dose of encouragement. Remember what it felt like  the first time you got it to work? :)

I feel similarly about visualizing. I think including a few simple plots could be a nice challenge that results in a win for the reader. On the other hand, building models feels different because it's not just about running the code. It requires a whole other set of tough concepts that bring meaning to that code. I think maybe we can do a linear regression model and that might be the most we can tackle. 

I really like the way things are coming along. I think it's totally fine to write and let this thing take some shape before we really start agonizing over details. Once the other parts of the narrative start to come together it will be easier to assess how well we met the goals of the narrative structure–or even if the original plan is still what we want. 

TL;DR (Just my thoughts, open to others)

**Do we need a chapter even as long as it is now, given the existence of other excellent resources (i.e., R4DS!)**: Yes

**Is the idea of downloading a CSV and then reading it useful, useful but a bit tedious, or not useful at this stage/it’s better to get started in some other ways:* Reading in data is useful for this chapter

**Should this chapter include visualizing and modeling data? communicating (w/ R Markdown) too?** Visualization, yes. Model, yes but only a little

**Comment by jrosen48** [comment]
Hi all, and thank you so much @kierisi, @ivelasq, and @restrellado for engaging with this and sharing very helpful ideas. This is fun! Okay, just to sum up a few things regarding next steps:
- I like the idea of delimiting this a bit - this aligns with things that all three of you said. Also, I really like the notion of embracing a project-based learning approach, with the walkthroughs describing a problem and walking through (one) solution. 
- Related, I think there's consensus that we stop short of visualizing and modeling here - and focus on setting up, accessing data, and doing some basic data processing/structuring. I concur! 

And here are what I see as three *things to do*
1. The idea of having two tracks seems to have some traction: I like this and forgot that *some* our readers will have experience using R - or at least may have used it a few times, attended a workshop, or been involved with a collaborator who used it. I think the following sounds great:
    - one can be 'installation and setup of R and R Studio', which covers installation, projects, and packages
    - the second can be 'data loading and manipulation using the tidyverse', which covers reading/saving files, pipes, selecting, filtering, etc.  chapters.
2. I concur that there's a lot of ways to load different data but not a descriptive of what to use when; I think we can and should address this. 
3. Finally, I think we should mention more about what purpose `swirl` or DataCamp may serve to our audience. 

We can keep this open or add these things to do as an issue. I'm in favor of the latter, if that sounds good.
